st2common/tests/unit/test_param_utils.py

```diff
@@ -362,15 +362,17 @@ def test_get_finalized_params_with_list(self):
             'a2': 'Test',
             'a3': 'Test2',
             'a4': '{{a1}}',
-            'a5': ['{{a2}}', '{{a3}}']
+            'a5': ['{{a2}}', '{{a3}}'],
+            'a6': [['{{r1}}', '{{a2}}'], ['{{a3}}', '{{a1}}']]
```

Comment:
It might be possible to capture the type of the object being rendered in to the template then cast to that, but that would only be relevant for single variable templates. We obviously can't cast `My variable is: 1` as an int. But if the template was `{{my_var}}` we can. I can determine this from a simple regex but we need to decide if it's worth it. I lean towards yes but I'm willing to get vetoed.